rpc/public-credentials/src/lib.rs

```diff
@@ -31,24 +31,30 @@ use sp_api::ProvideRuntimeApi;
 use sp_blockchain::HeaderBackend;
 use sp_runtime::{generic::BlockId, traits::Block as BlockT};
 
+/// Filter that can be used after the credentials for a given subject have been
+/// retrieved from the blockchain state.
+pub trait PublicCredentialsFilter<Credential> {
```

Comment:
There is a [trait in substrate that does this](https://paritytech.github.io/substrate/master/frame_support/traits/trait.Contains.html). This is used everywhere where they want to filter somethings. I think we should follow this even tho the naming of [`Everything`](https://paritytech.github.io/substrate/master/frame_support/traits/enum.Everything.html) and [`Nothing`](https://paritytech.github.io/substrate/master/frame_support/traits/enum.Nothing.html) is a bit confusing.

Reply:
This does not work in our case. The trait has an associated function and does not take `&self` as a parameter, while we need to know both the values being filtered (from the retrieved collection) and the one that we want to filter by (the passed filter).

Reply:
Giving the value to filter in the parameter also would not work, since we could have two different properties that are of the same type, e.g., `Hash`, and the filter would not be able to tell the two apart.

Reply:
ah ok didn't notice that